src/Rule/Ip.php

```diff
@@ -249,41 +248,82 @@ public function getRanges()
         return $this->ranges;
     }
 
-    protected function validateValue($value): Result
+    /**
+     * @param bool $ipv6
+     *
+     * @return Ip
+     */
+    public function useIpv6(bool $ipv6): self
     {
-        $result = new Result();
+        $this->ipv6 = $ipv6;
+
+        return $this;
+    }
+
+    /**
+     * @param bool $ipv4
+     *
+     * @return Ip
+     */
+    public function useIpv4($ipv4): self
+    {
+        $this->ipv4 = $ipv4;
+
+        return $this;
+    }
 
+    /**
+     * @param bool $subnet
+     *
+     * @return Ip
+     */
+    public function useSubnet($subnet): self
```

Comment:
I'd introduce two methods instead:

- requireSubnet
- optionalSubnet

Also internally it makes sense to introduce three private constants instead of having true/false/null as magic values.